### PR TITLE
Move gh client to server struct.

### PR DESCRIPTION
**Pull request by metanerd**
#### Summary
Fix so that not a new github client is created before every github request. 

#### Ticket Link

### CODE REVIEW

File: server/auto_merge.go [hunk 1]
```diff
@@ -6,7 +6,6 @@ package server
 import (
 	"context"
 	"fmt"
-
```
Comment (hanzei) [inline server/auto_merge.go:9]: It's idiomatic for go code to group all imports from the `std` lib together. This is also the style that https://pkg.go.dev/golang.org/x/tools/cmd/goimports?tab=doc enforces.

Why drop this style?

File: server/limit_rate_gh.go [hunk 2]
```diff
@@ -12,9 +12,7 @@ import (
 
 func (s *Server) CheckLimitRateAndSleep() {
 	mlog.Info("Checking the rate limit on Github and will sleep if need...")
-
```
Comment (hanzei) [inline server/limit_rate_gh.go:15]: Nit: Why is this line getting dropped?

**Comment by hanzei** [comment]
Nice cleanup :+1:

File: server/limit_rate_gh.go (hunk 2 above)
Comment (metanerd) [reply server/limit_rate_gh.go:15]: The line didn't make the code any more readable in my opinion.

File: server/auto_merge.go (hunk 1 above)
Comment (metanerd) [reply server/auto_merge.go]: I didn't know about this tool. Makes sense to add it to the CI. I added it.

Comment (hanzei) [reply server/auto_merge.go]: I don't have any objections on using `goimports`. Other Mattermost project using `golangci-lint` as a metalinter. Hence, it makes sense to also integrate `golangci-lint` into this repo, which can also run `goimports`.

**Pull request merged**